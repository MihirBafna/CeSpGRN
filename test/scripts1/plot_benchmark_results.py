# In[0]
from numpy.lib.function_base import disp
import pandas as pd
import numpy as np
import torch
import torch.nn
import matplotlib.pyplot as plt
import sys, os
sys.path.append('../../src/')

import matplotlib.pyplot as plt
import seaborn as sns
from scipy.spatial.distance import pdist, squareform

import bmk_beeline as bmk
import warnings
warnings.filterwarnings("ignore")

from matplotlib import rcParams

labelsize = 16
rcParams['xtick.labelsize'] = labelsize
rcParams['ytick.labelsize'] = labelsize 


# TODO: Compare between models, either using the best score or the average score, check if there are significant differences between models
# In[1]
ntimes = 3000
interval = 200
ngenes = 20
result_dir = "../results/GGM_" + str(ntimes) + "_" + str(interval) + "_" + str(ngenes) + "/"
# baseline scores
scores_random = pd.read_csv(result_dir + "score_rand.csv")

# model 1, select the one with the highest median value
alpha = 1
rho = "adaptive"
best_nmse = [100, 0, 0]
best_pearson = [0, 0, 0]
best_cos = [0, 0, 0]
for bandwidth in [0.01, 0.1, 1, 10]:
    for lamb in [0.01, 0.1, 1]:
        data = str(bandwidth) + "_" + str(alpha) + "_" + str(lamb) + "_" + rho
        score = pd.read_csv(result_dir + "score_" + data + ".csv")
        # extract median score across times
        nmse_med = np.median(score["nmse"].values)
        # pos_med = np.median(score["probability of success"].values)
        pearson_med = np.median([eval(x)[0] for x in score["pearson"].values])
        cos_med = np.median(score["cosine similarity"].values)
        print("bandwidth: " + str(bandwidth) + ", alpha: " + str(alpha) + ", lambda: " + str(lamb) + ", rho: " + str(rho))
        print("nmse: " + str(nmse_med) + ", pearson: " + str(pearson_med) + ", cos simi: " + str(cos_med))
        if nmse_med < best_nmse[0]:
            best_nmse[0] = nmse_med
            best_nmse[1] = bandwidth
            best_nmse[2] = lamb
        if pearson_med > best_pearson[0]:
            best_pearson[0] = pearson_med
            best_pearson[1] = bandwidth
            best_pearson[2] = lamb
        if cos_med > best_cos[0]:
            best_cos[0] = cos_med
            best_cos[1] = bandwidth
            best_cos[2] = lamb
print()
print("best nmse: " + str(best_nmse[0]) + ", bandwidth: " + str(best_nmse[1]) + ", lamb: " + str(best_nmse[2]))
print("best pearson: " + str(best_pearson[0]) + ", bandwidth: " + str(best_pearson[1]) + ", lamb: " + str(best_pearson[2]))
print("best cos: " + str(best_cos[0]) + ", bandwidth: " + str(best_cos[1]) + ", lamb: " + str(best_cos[2]))
print()

fig = plt.figure(figsize = (10, 21))
ax = fig.subplots(nrows = 3, ncols = 1)
data = str(best_nmse[1]) + "_" + str(alpha) + "_" + str(best_nmse[2]) + "_" + rho
score = pd.read_csv(result_dir + "score_" + data + ".csv")
scores_random["pearson"] = [eval(x)[0] for x in scores_random["pearson"].values]
score["pearson"] = [eval(x)[0] for x in score["pearson"].values]
score_full = pd.concat([score, scores_random], axis = 0)
sns.boxplot(data = score_full, x = "model", y = "nmse", ax = ax[0])
sns.boxplot(data = score_full, x = "model", y = "pearson", ax = ax[1])
sns.boxplot(data = score_full, x = "model", y = "cosine similarity", ax = ax[2])
fig.savefig(result_dir + "score_boxplot1.png", bbox_inches = "tight")

# In[2]

# model 2
alpha = 2
rho = "1.7"
best_nmse = [100, 0, 0]
best_pearson = [0, 0, 0]
best_cos = [0, 0, 0]
for bandwidth in [0.01, 0.1, 1, 10]:
    for lamb in [0.01, 0.1, 1]:
        data = str(bandwidth) + "_" + str(alpha) + "_" + str(lamb) + "_" + rho
        score = pd.read_csv(result_dir + "score_" + data + ".csv")
        # extract median score across times
        nmse_med = np.median(score["nmse"].values)
        # pos_med = np.median(score["probability of success"].values)
        pearson_med = np.median([eval(x)[0] for x in score["pearson"].values])
        cos_med = np.median(score["cosine similarity"].values)
        print("bandwidth: " + str(bandwidth) + ", alpha: " + str(alpha) + ", lambda: " + str(lamb) + ", rho: " + str(rho))
        print("nmse: " + str(nmse_med) + ", pearson: " + str(pearson_med) + ", cos simi: " + str(cos_med))
        if nmse_med < best_nmse[0]:
            best_nmse[0] = nmse_med
            best_nmse[1] = bandwidth
            best_nmse[2] = lamb
        if pearson_med > best_pearson[0]:
            best_pearson[0] = pearson_med
            best_pearson[1] = bandwidth
            best_pearson[2] = lamb
        if cos_med > best_cos[0]:
            best_cos[0] = cos_med
            best_cos[1] = bandwidth
            best_cos[2] = lamb
print()
print("best nmse: " + str(best_nmse[0]) + ", bandwidth: " + str(best_nmse[1]) + ", lamb: " + str(best_nmse[2]))
print("best pearson: " + str(best_pearson[0]) + ", bandwidth: " + str(best_pearson[1]) + ", lamb: " + str(best_pearson[2]))
print("best cos: " + str(best_cos[0]) + ", bandwidth: " + str(best_cos[1]) + ", lamb: " + str(best_cos[2]))
print()
fig = plt.figure(figsize = (10, 21))
ax = fig.subplots(nrows = 3, ncols = 1)
data = str(best_nmse[1]) + "_" + str(alpha) + "_" + str(best_nmse[2]) + "_" + rho
score = pd.read_csv(result_dir + "score_" + data + ".csv")
score["pearson"] = [eval(x)[0] for x in score["pearson"].values]
score_full = pd.concat([score, scores_random], axis = 0)
sns.boxplot(data = score_full, x = "model", y = "nmse", ax = ax[0])
sns.boxplot(data = score_full, x = "model", y = "pearson", ax = ax[1])
sns.boxplot(data = score_full, x = "model", y = "cosine similarity", ax = ax[2])
fig.savefig(result_dir + "score_boxplot2.png", bbox_inches = "tight")

# In[2]
# model 3
alpha = 1
rho = "1.7"
best_nmse = [100, 0, 0]
best_pearson = [0, 0, 0]
best_cos = [0, 0, 0]
for bandwidth in [0.01, 0.1, 1, 10]:
    for lamb in [0.01, 0.1, 1]:
        data = str(bandwidth) + "_" + str(alpha) + "_" + str(lamb) + "_" + rho
        score = pd.read_csv(result_dir + "score_" + data + ".csv")
        # extract median score across times
        nmse_med = np.median(score["nmse"].values)
        # pos_med = np.median(score["probability of success"].values)
        pearson_med = np.median([eval(x)[0] for x in score["pearson"].values])
        cos_med = np.median(score["cosine similarity"].values)
        print("bandwidth: " + str(bandwidth) + ", alpha: " + str(alpha) + ", lambda: " + str(lamb) + ", rho: " + str(rho))
        print("nmse: " + str(nmse_med) + ", pearson: " + str(pearson_med) + ", cos simi: " + str(cos_med))
        if nmse_med < best_nmse[0]:
            best_nmse[0] = nmse_med
            best_nmse[1] = bandwidth
            best_nmse[2] = lamb
        if pearson_med > best_pearson[0]:
            best_pearson[0] = pearson_med
            best_pearson[1] = bandwidth
            best_pearson[2] = lamb
        if cos_med > best_cos[0]:
            best_cos[0] = cos_med
            best_cos[1] = bandwidth
            best_cos[2] = lamb
print()
print("best nmse: " + str(best_nmse[0]) + ", bandwidth: " + str(best_nmse[1]) + ", lamb: " + str(best_nmse[2]))
print("best pearson: " + str(best_pearson[0]) + ", bandwidth: " + str(best_pearson[1]) + ", lamb: " + str(best_pearson[2]))
print("best cos: " + str(best_cos[0]) + ", bandwidth: " + str(best_cos[1]) + ", lamb: " + str(best_cos[2]))
print()
print()
print("model using no mean")
print()
fig = plt.figure(figsize = (10, 21))
ax = fig.subplots(nrows = 3, ncols = 1)
data = str(best_nmse[1]) + "_" + str(alpha) + "_" + str(best_nmse[2]) + "_" + rho
score = pd.read_csv(result_dir + "score_" + data + ".csv")
score["pearson"] = [eval(x)[0] for x in score["pearson"].values]
score_full = pd.concat([score, scores_random], axis = 0)
sns.boxplot(data = score_full, x = "model", y = "nmse", ax = ax[0])
sns.boxplot(data = score_full, x = "model", y = "pearson", ax = ax[1])
sns.boxplot(data = score_full, x = "model", y = "cosine similarity", ax = ax[2])
fig.savefig(result_dir + "score_boxplot3.png", bbox_inches = "tight")

# In[2]
# model 4
alpha = 1
rho = "adaptive"
best_nmse = [100, 0, 0]
best_pearson = [0, 0, 0]
best_cos = [0, 0, 0]
for bandwidth in [0.01, 0.1, 1, 10]:
    for lamb in [0.01, 0.1, 1]:
        data = str(bandwidth) + "_" + str(alpha) + "_" + str(lamb) + "_" + rho
        score = pd.read_csv(result_dir + "score_" + data + "_nomean.csv")
        # extract median score across times
        nmse_med = np.median(score["nmse"].values)
        # pos_med = np.median(score["probability of success"].values)
        pearson_med = np.median([eval(x)[0] for x in score["pearson"].values])
        cos_med = np.median(score["cosine similarity"].values)
        print("bandwidth: " + str(bandwidth) + ", alpha: " + str(alpha) + ", lambda: " + str(lamb) + ", rho: " + str(rho))
        print("nmse: " + str(nmse_med) + ", pearson: " + str(pearson_med) + ", cos simi: " + str(cos_med))
        if nmse_med < best_nmse[0]:
            best_nmse[0] = nmse_med
            best_nmse[1] = bandwidth
            best_nmse[2] = lamb
        if pearson_med > best_pearson[0]:
            best_pearson[0] = pearson_med
            best_pearson[1] = bandwidth
            best_pearson[2] = lamb
        if cos_med > best_cos[0]:
            best_cos[0] = cos_med
            best_cos[1] = bandwidth
            best_cos[2] = lamb
print()
print("best nmse: " + str(best_nmse[0]) + ", bandwidth: " + str(best_nmse[1]) + ", lamb: " + str(best_nmse[2]))
print("best pearson: " + str(best_pearson[0]) + ", bandwidth: " + str(best_pearson[1]) + ", lamb: " + str(best_pearson[2]))
print("best cos: " + str(best_cos[0]) + ", bandwidth: " + str(best_cos[1]) + ", lamb: " + str(best_cos[2]))
print()
fig = plt.figure(figsize = (10, 21))
ax = fig.subplots(nrows = 3, ncols = 1)
data = str(best_nmse[1]) + "_" + str(alpha) + "_" + str(best_nmse[2]) + "_" + rho
score = pd.read_csv(result_dir + "score_" + data + "_nomean.csv")
score["pearson"] = [eval(x)[0] for x in score["pearson"].values]
score_full = pd.concat([score, scores_random], axis = 0)
sns.boxplot(data = score_full, x = "model", y = "nmse", ax = ax[0])
sns.boxplot(data = score_full, x = "model", y = "pearson", ax = ax[1])
sns.boxplot(data = score_full, x = "model", y = "cosine similarity", ax = ax[2])
fig.savefig(result_dir + "score_boxplot4.png", bbox_inches = "tight")


# In[2]
# model 5
alpha = 2
rho = "1.7"
best_nmse = [100, 0, 0]
best_pearson = [0, 0, 0]
best_cos = [0, 0, 0]
for bandwidth in [0.01, 0.1, 1, 10]:
    for lamb in [0.01, 0.1, 1]:
        data = str(bandwidth) + "_" + str(alpha) + "_" + str(lamb) + "_" + rho
        score = pd.read_csv(result_dir + "score_" + data + "_nomean.csv")
        # extract median score across times
        nmse_med = np.median(score["nmse"].values)
        # pos_med = np.median(score["probability of success"].values)
        pearson_med = np.median([eval(x)[0] for x in score["pearson"].values])
        cos_med = np.median(score["cosine similarity"].values)
        print("bandwidth: " + str(bandwidth) + ", alpha: " + str(alpha) + ", lambda: " + str(lamb) + ", rho: " + str(rho))
        print("nmse: " + str(nmse_med) + ", pearson: " + str(pearson_med) + ", cos simi: " + str(cos_med))
        if nmse_med < best_nmse[0]:
            best_nmse[0] = nmse_med
            best_nmse[1] = bandwidth
            best_nmse[2] = lamb
        if pearson_med > best_pearson[0]:
            best_pearson[0] = pearson_med
            best_pearson[1] = bandwidth
            best_pearson[2] = lamb
        if cos_med > best_cos[0]:
            best_cos[0] = cos_med
            best_cos[1] = bandwidth
            best_cos[2] = lamb
print()
print("best nmse: " + str(best_nmse[0]) + ", bandwidth: " + str(best_nmse[1]) + ", lamb: " + str(best_nmse[2]))
print("best pearson: " + str(best_pearson[0]) + ", bandwidth: " + str(best_pearson[1]) + ", lamb: " + str(best_pearson[2]))
print("best cos: " + str(best_cos[0]) + ", bandwidth: " + str(best_cos[1]) + ", lamb: " + str(best_cos[2]))
print()
fig = plt.figure(figsize = (10, 21))
ax = fig.subplots(nrows = 3, ncols = 1)
data = str(best_nmse[1]) + "_" + str(alpha) + "_" + str(best_nmse[2]) + "_" + rho
score = pd.read_csv(result_dir + "score_" + data + "_nomean.csv")
score["pearson"] = [eval(x)[0] for x in score["pearson"].values]
score_full = pd.concat([score, scores_random], axis = 0)
sns.boxplot(data = score_full, x = "model", y = "nmse", ax = ax[0])
sns.boxplot(data = score_full, x = "model", y = "pearson", ax = ax[1])
sns.boxplot(data = score_full, x = "model", y = "cosine similarity", ax = ax[2])
fig.savefig(result_dir + "score_boxplot5.png", bbox_inches = "tight")

# model 6
alpha = 1
rho = "1.7"
best_nmse = [100, 0, 0]
best_pearson = [0, 0, 0]
best_cos = [0, 0, 0]
for bandwidth in [0.01, 0.1, 1, 10]:
    for lamb in [0.01, 0.1, 1]:
        data = str(bandwidth) + "_" + str(alpha) + "_" + str(lamb) + "_" + rho
        score = pd.read_csv(result_dir + "score_" + data + "_nomean.csv")
        # extract median score across times
        nmse_med = np.median(score["nmse"].values)
        # pos_med = np.median(score["probability of success"].values)
        pearson_med = np.median([eval(x)[0] for x in score["pearson"].values])
        cos_med = np.median(score["cosine similarity"].values)
        print("bandwidth: " + str(bandwidth) + ", alpha: " + str(alpha) + ", lambda: " + str(lamb) + ", rho: " + str(rho))
        print("nmse: " + str(nmse_med) + ", pearson: " + str(pearson_med) + ", cos simi: " + str(cos_med))
        if nmse_med < best_nmse[0]:
            best_nmse[0] = nmse_med
            best_nmse[1] = bandwidth
            best_nmse[2] = lamb
        if pearson_med > best_pearson[0]:
            best_pearson[0] = pearson_med
            best_pearson[1] = bandwidth
            best_pearson[2] = lamb
        if cos_med > best_cos[0]:
            best_cos[0] = cos_med
            best_cos[1] = bandwidth
            best_cos[2] = lamb
print()
print("best nmse: " + str(best_nmse[0]) + ", bandwidth: " + str(best_nmse[1]) + ", lamb: " + str(best_nmse[2]))
print("best pearson: " + str(best_pearson[0]) + ", bandwidth: " + str(best_pearson[1]) + ", lamb: " + str(best_pearson[2]))
print("best cos: " + str(best_cos[0]) + ", bandwidth: " + str(best_cos[1]) + ", lamb: " + str(best_cos[2]))
print()
fig = plt.figure(figsize = (10, 21))
ax = fig.subplots(nrows = 3, ncols = 1)
data = str(best_nmse[1]) + "_" + str(alpha) + "_" + str(best_nmse[2]) + "_" + rho
score = pd.read_csv(result_dir + "score_" + data + "_nomean.csv")
score["pearson"] = [eval(x)[0] for x in score["pearson"].values]
score_full = pd.concat([score, scores_random], axis = 0)
sns.boxplot(data = score_full, x = "model", y = "nmse", ax = ax[0])
sns.boxplot(data = score_full, x = "model", y = "pearson", ax = ax[1])
sns.boxplot(data = score_full, x = "model", y = "cosine similarity", ax = ax[2])
fig.savefig(result_dir + "score_boxplot6.png", bbox_inches = "tight")

# %%
