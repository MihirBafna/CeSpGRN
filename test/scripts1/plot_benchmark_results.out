---------------------------------------
Begin PBS Prologue Tue Sep 21 22:35:27 EDT 2021
Job ID:     7016762.sched-hive.pace.gatech.edu
User ID:    zzhang834
Job name:   plot_benchmark_results
Queue:      hive-himem
End PBS Prologue Tue Sep 21 22:35:27 EDT 2021
---------------------------------------
bandwidth: 0.01, alpha: 1, lambda: 0.01, rho: adaptive
nmse: 0.937234014333546, pearson: 0.9617950265093597, cos simi: 0.25218426709107766
bandwidth: 0.01, alpha: 1, lambda: 0.1, rho: adaptive
nmse: 0.9577207199903623, pearson: 0.9169310311247925, cos simi: 0.2316711380590544
bandwidth: 0.01, alpha: 1, lambda: 1, rho: adaptive
nmse: 0.985880957848791, pearson: 0.8873434150048009, cos simi: 0.22004505336713887
bandwidth: 0.1, alpha: 1, lambda: 0.01, rho: adaptive
nmse: 0.935421365261775, pearson: 0.9720529457543567, cos simi: 0.25792770976443435
bandwidth: 0.1, alpha: 1, lambda: 0.1, rho: adaptive
nmse: 0.9577315022742541, pearson: 0.9174513681554555, cos simi: 0.23278612073637625
bandwidth: 0.1, alpha: 1, lambda: 1, rho: adaptive
nmse: 0.985917909030495, pearson: 0.8870197542896475, cos simi: 0.22001574343567815
bandwidth: 1, alpha: 1, lambda: 0.01, rho: adaptive
nmse: 0.9392700310360206, pearson: 0.9677284730919222, cos simi: 0.251740322639431
bandwidth: 1, alpha: 1, lambda: 0.1, rho: adaptive
nmse: 0.958004688008414, pearson: 0.911249593291211, cos simi: 0.23445896639651545
bandwidth: 1, alpha: 1, lambda: 1, rho: adaptive
nmse: 0.9859411204799373, pearson: 0.8876051147843538, cos simi: 0.22011186213965572
bandwidth: 10, alpha: 1, lambda: 0.01, rho: adaptive
nmse: 0.9404530940268809, pearson: 0.9524299384033845, cos simi: 0.2508699590122251
bandwidth: 10, alpha: 1, lambda: 0.1, rho: adaptive
nmse: 0.9583702363160802, pearson: 0.9066463371128923, cos simi: 0.23569761377033024
bandwidth: 10, alpha: 1, lambda: 1, rho: adaptive
nmse: 0.9860890122857716, pearson: 0.8886639024558363, cos simi: 0.22024759127954885

best nmse: 0.935421365261775, bandwidth: 0.1, lamb: 0.01
best pearson: 0.9720529457543567, bandwidth: 0.1, lamb: 0.01
best cos: 0.25792770976443435, bandwidth: 0.1, lamb: 0.01

bandwidth: 0.01, alpha: 2, lambda: 0.01, rho: 1.7
nmse: 0.0400048961275328, pearson: 0.9807953834191716, cos simi: 0.9817364706429865
bandwidth: 0.01, alpha: 2, lambda: 0.1, rho: 1.7
nmse: 0.29782255495457743, pearson: 0.936568382931394, cos simi: 0.9397201165955704
bandwidth: 0.01, alpha: 2, lambda: 1, rho: 1.7
nmse: 0.7604643114730494, pearson: 0.9032943213526393, cos simi: 0.9081311707853719
bandwidth: 0.1, alpha: 2, lambda: 0.01, rho: 1.7
nmse: 0.02972370894952345, pearson: 0.9911972069535555, cos simi: 0.991620439641482
bandwidth: 0.1, alpha: 2, lambda: 0.1, rho: 1.7
nmse: 0.30439731799381625, pearson: 0.9358633520649479, cos simi: 0.9390941495705167
bandwidth: 0.1, alpha: 2, lambda: 1, rho: 1.7
nmse: 0.7623580212513046, pearson: 0.9034283974871845, cos simi: 0.908313979627875
bandwidth: 1, alpha: 2, lambda: 0.01, rho: 1.7
nmse: 0.0491544533426448, pearson: 0.986391776016241, cos simi: 0.9869908682449209
bandwidth: 1, alpha: 2, lambda: 0.1, rho: 1.7
nmse: 0.3189239958808911, pearson: 0.9306668797842288, cos simi: 0.9340823995453305
bandwidth: 1, alpha: 2, lambda: 1, rho: 1.7
nmse: 0.7629699026086739, pearson: 0.9041572769200872, cos simi: 0.9089766475142969
bandwidth: 10, alpha: 2, lambda: 0.01, rho: 1.7
nmse: 0.08939613459181135, pearson: 0.9690707043361413, cos simi: 0.9706125374781681
bandwidth: 10, alpha: 2, lambda: 0.1, rho: 1.7
nmse: 0.33167778921874014, pearson: 0.9264893258778726, cos simi: 0.9299511203522877
bandwidth: 10, alpha: 2, lambda: 1, rho: 1.7
nmse: 0.7633024911719015, pearson: 0.9043445827737555, cos simi: 0.9091228249106662

best nmse: 0.02972370894952345, bandwidth: 0.1, lamb: 0.01
best pearson: 0.9911972069535555, bandwidth: 0.1, lamb: 0.01
best cos: 0.991620439641482, bandwidth: 0.1, lamb: 0.01

bandwidth: 0.01, alpha: 1, lambda: 0.01, rho: 1.7
nmse: 0.04000515578392975, pearson: 0.9807953475504885, cos simi: 0.9817364223578362
bandwidth: 0.01, alpha: 1, lambda: 0.1, rho: 1.7
nmse: 0.2978230231024588, pearson: 0.9365684722026277, cos simi: 0.9397201936160848
bandwidth: 0.01, alpha: 1, lambda: 1, rho: 1.7
nmse: 0.7604643009260399, pearson: 0.9032943194858929, cos simi: 0.908131199308122
bandwidth: 0.1, alpha: 1, lambda: 0.01, rho: 1.7
nmse: 0.0297242368404822, pearson: 0.9911971754531168, cos simi: 0.991620423045964
bandwidth: 0.1, alpha: 1, lambda: 0.1, rho: 1.7
nmse: 0.3043977755467874, pearson: 0.9358634636203336, cos simi: 0.9390942488780822
bandwidth: 0.1, alpha: 1, lambda: 1, rho: 1.7
nmse: 0.7623580007449047, pearson: 0.9034283989657743, cos simi: 0.9083140104142011
bandwidth: 1, alpha: 1, lambda: 0.01, rho: 1.7
nmse: 0.0491549077881871, pearson: 0.9863917829094565, cos simi: 0.9869908603815004
bandwidth: 1, alpha: 1, lambda: 0.1, rho: 1.7
nmse: 0.3189244079301953, pearson: 0.9306670214479205, cos simi: 0.9340825197829608
bandwidth: 1, alpha: 1, lambda: 1, rho: 1.7
nmse: 0.7629698933519466, pearson: 0.9041572771689304, cos simi: 0.908976608210815
bandwidth: 10, alpha: 1, lambda: 0.01, rho: 1.7
nmse: 0.0893963709906695, pearson: 0.9690707069298597, cos simi: 0.97061255573107
bandwidth: 10, alpha: 1, lambda: 0.1, rho: 1.7
nmse: 0.33167822389434, pearson: 0.9264894439184616, cos simi: 0.9299511798348514
bandwidth: 10, alpha: 1, lambda: 1, rho: 1.7
nmse: 0.7633024768368124, pearson: 0.9043445835585359, cos simi: 0.9091228249479304

best nmse: 0.0297242368404822, bandwidth: 0.1, lamb: 0.01
best pearson: 0.9911971754531168, bandwidth: 0.1, lamb: 0.01
best cos: 0.991620423045964, bandwidth: 0.1, lamb: 0.01


model using no mean

bandwidth: 0.01, alpha: 1, lambda: 0.01, rho: adaptive
nmse: 50.408994133540105, pearson: -0.03621281024291742, cos simi: -0.034198406899976955
bandwidth: 0.01, alpha: 1, lambda: 0.1, rho: adaptive
nmse: 16.69038392579521, pearson: -0.015730436672891347, cos simi: -0.00997084650749545
bandwidth: 0.01, alpha: 1, lambda: 1, rho: adaptive
nmse: 2.3072565874362727, pearson: 0.0024794887545633616, cos simi: 0.0
bandwidth: 0.1, alpha: 1, lambda: 0.01, rho: adaptive
nmse: 45.89938171591767, pearson: -0.02551718109997386, cos simi: -0.02144419136615385
bandwidth: 0.1, alpha: 1, lambda: 0.1, rho: adaptive
nmse: 16.17689496165074, pearson: -0.008619426666594779, cos simi: -0.0064457062322782
bandwidth: 0.1, alpha: 1, lambda: 1, rho: adaptive
nmse: 2.216429007516086, pearson: 0.0024792564888100888, cos simi: 0.0
bandwidth: 1, alpha: 1, lambda: 0.01, rho: adaptive
nmse: 41.57848692781587, pearson: -0.01729910189157502, cos simi: -0.02286265443818555
bandwidth: 1, alpha: 1, lambda: 0.1, rho: adaptive
nmse: 15.262110337620289, pearson: -0.005833267587489293, cos simi: -0.0063988179213731
bandwidth: 1, alpha: 1, lambda: 1, rho: adaptive
nmse: 2.176601460073889, pearson: 0.0024772254267849524, cos simi: 0.0
bandwidth: 10, alpha: 1, lambda: 0.01, rho: adaptive
nmse: 37.71832279266362, pearson: -0.024511468797336837, cos simi: -0.01890279887500395
bandwidth: 10, alpha: 1, lambda: 0.1, rho: adaptive
nmse: 14.850211939969721, pearson: -0.004850672163060429, cos simi: 0.0
bandwidth: 10, alpha: 1, lambda: 1, rho: adaptive
nmse: 2.185981555911253, pearson: 0.0024742690006078673, cos simi: 0.0

best nmse: 2.176601460073889, bandwidth: 1, lamb: 1
best pearson: 0.0024794887545633616, bandwidth: 0.01, lamb: 1
best cos: 0, bandwidth: 0, lamb: 0

bandwidth: 0.01, alpha: 2, lambda: 0.01, rho: 1.7
nmse: 198.0287939659787, pearson: -0.036283569412370896, cos simi: -0.034205610602748154
bandwidth: 0.01, alpha: 2, lambda: 0.1, rho: 1.7
nmse: 63.60491147940696, pearson: -0.015682475948725554, cos simi: -0.010208623840096
bandwidth: 0.01, alpha: 2, lambda: 1, rho: 1.7
nmse: 5.755020989258054, pearson: 0.0024776610852561766, cos simi: 0.0
bandwidth: 0.1, alpha: 2, lambda: 0.01, rho: 1.7
nmse: 180.2219173527761, pearson: -0.025510335875525002, cos simi: -0.021441989196449497
bandwidth: 0.1, alpha: 2, lambda: 0.1, rho: 1.7
nmse: 61.53341276646298, pearson: -0.008731237512748763, cos simi: -0.0066451106154541
bandwidth: 0.1, alpha: 2, lambda: 1, rho: 1.7
nmse: 5.699510801016897, pearson: 0.0024771032569349738, cos simi: 0.0
bandwidth: 1, alpha: 2, lambda: 0.01, rho: 1.7
nmse: 162.99686079149828, pearson: -0.017291906477992586, cos simi: -0.022862680843689802
bandwidth: 1, alpha: 2, lambda: 0.1, rho: 1.7
nmse: 58.330499885744786, pearson: -0.0060168510085279025, cos simi: -0.0064512835949738
bandwidth: 1, alpha: 2, lambda: 1, rho: 1.7
nmse: 5.6701946784095405, pearson: 0.002477112339786379, cos simi: 0.0
bandwidth: 10, alpha: 2, lambda: 0.01, rho: 1.7
nmse: 147.88503141185072, pearson: -0.024527463459657388, cos simi: -0.01889496597849595
bandwidth: 10, alpha: 2, lambda: 0.1, rho: 1.7
nmse: 56.30832330116645, pearson: -0.005530385201696642, cos simi: 0.0
bandwidth: 10, alpha: 2, lambda: 1, rho: 1.7
nmse: 5.650074311190393, pearson: 0.002473453210612902, cos simi: 0.0

best nmse: 5.650074311190393, bandwidth: 10, lamb: 1
best pearson: 0.0024776610852561766, bandwidth: 0.01, lamb: 1
best cos: 0, bandwidth: 0, lamb: 0

bandwidth: 0.01, alpha: 1, lambda: 0.01, rho: 1.7
nmse: 198.02790710791442, pearson: -0.03628358771930504, cos simi: -0.0342056300994559
bandwidth: 0.01, alpha: 1, lambda: 0.1, rho: 1.7
nmse: 63.6047324769882, pearson: -0.01568249817835375, cos simi: -0.0102086243577781
bandwidth: 0.01, alpha: 1, lambda: 1, rho: 1.7
nmse: 5.755021781662196, pearson: 0.00247766107617295, cos simi: 0.0
bandwidth: 0.1, alpha: 1, lambda: 0.01, rho: 1.7
nmse: 180.22106353766654, pearson: -0.025510330384073843, cos simi: -0.0214420180687729
bandwidth: 0.1, alpha: 1, lambda: 0.1, rho: 1.7
nmse: 61.53324470050214, pearson: -0.008731243178354602, cos simi: -0.00664511051023575
bandwidth: 0.1, alpha: 1, lambda: 1, rho: 1.7
nmse: 5.699511488383903, pearson: 0.0024771032563106754, cos simi: 0.0
bandwidth: 1, alpha: 1, lambda: 0.01, rho: 1.7
nmse: 162.99632297846634, pearson: -0.017291913765227124, cos simi: -0.0228627031053157
bandwidth: 1, alpha: 1, lambda: 0.1, rho: 1.7
nmse: 58.33034167087678, pearson: -0.006016862502340967, cos simi: -0.006451282255056199
bandwidth: 1, alpha: 1, lambda: 1, rho: 1.7
nmse: 5.670195512148984, pearson: 0.002477112337685762, cos simi: 0.0
bandwidth: 10, alpha: 1, lambda: 0.01, rho: 1.7
nmse: 147.88470508292642, pearson: -0.02452747357562231, cos simi: -0.01889496622141975
bandwidth: 10, alpha: 1, lambda: 0.1, rho: 1.7
nmse: 56.308184212967205, pearson: -0.005530394965270123, cos simi: 0.0
bandwidth: 10, alpha: 1, lambda: 1, rho: 1.7
nmse: 5.650075003992775, pearson: 0.00247345320999437, cos simi: 0.0

best nmse: 5.650075003992775, bandwidth: 10, lamb: 1
best pearson: 0.00247766107617295, bandwidth: 0.01, lamb: 1
best cos: 0, bandwidth: 0, lamb: 0

---------------------------------------
Begin PBS Epilogue Tue Sep 21 22:35:35 EDT 2021
Job ID:     7016762.sched-hive.pace.gatech.edu
User ID:    zzhang834
Job name:   plot_benchmark_results
Resources:  nodes=1:ppn=4,walltime=36:20:00,neednodes=1:ppn=4
Rsrc Used:  cput=00:00:08,vmem=0kb,walltime=00:00:09,mem=0kb,energy_used=0
Queue:      hive-himem
Nodes:     
atl1-1-01-018-38.pace.gatech.edu
End PBS Epilogue Tue Sep 21 22:35:35 EDT 2021
---------------------------------------
